### PR TITLE
dx: conventional message feature added

### LINKED ISSUE #3: [DX] configure commitlint with husky
As discussed

### PR DESCRIPTION
**Pull request by dhruvrajsinh-7**
closes #3

### CODE REVIEW

File: .commitlintrc.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": ["@commitlint/config-conventional"],
+    "rules": {
+        "type-enum": [2, "always", [
+            "build", "chore", "ci", "docs", "feat", "fix", "perf", "refactor", "revert", "style", "test"
```
Comment (Nabhag8848) [inline .commitlintrc.json:8]: add: `dx` (developer experience) as allowed convention

File: package.json [hunk 2]
```diff
@@ -24,10 +24,13 @@
     "winston": "^3.8.2"
   },
```
Comment (Nabhag8848) [inline package.json:26]: add `"postinstall": "husky install"` under scripts objects to solve the husky not installed when installing the dependency

**Review by Nabhag8848 — CHANGES_REQUESTED** [review]
Lets fly you are almost there.

File: package.json (hunk 2 above)
Comment (dhruvrajsinh-7) [reply package.json:26]: fixed the issue of dependency check it now

**Review by Nabhag8848 — APPROVED** [review]
# LGTM!

**Pull request merged**